src/ConversationList/ConversationListBox.vala

```diff
@@ -425,4 +451,62 @@ public class Mail.ConversationListBox : VirtualizingListBox {
     public void undo_expired () {
         move_handler.expire_undo ();
     }
+
+    private bool create_context_menu (Gdk.Event e, ConversationListItem row) {
+        var item = (ConversationItemModel)row.model_item;
+
+        var menu = new Gtk.Menu ();
+
+        var delete_menu_item = new Gtk.MenuItem.with_label (_("Delete"));
+        menu.add (delete_menu_item);
+
+        delete_menu_item.activate.connect (() => {
+            trash_selected_messages ();
+        });
+
+        if (!item.unread) {
+            var mark_unread_menu_item = new Gtk.MenuItem.with_label (_("Mark As Unread"));
+            menu.add (mark_unread_menu_item);
```

Comment:
Making the keyboard shortcut discoverable here is probably a good idea as well :)